### PR TITLE
fix: add permission name when accessing a special file errors

### PR DESCRIPTION
**Pull request by sigmaSd**
fix https://github.com/denoland/deno/issues/25055

### CODE REVIEW

**Review by dsherret — COMMENTED** [review]
Can you add a test to ensure it's actually fixed and doesn't regress in the future?

**Comment by sigmaSd** [comment]
I found something weird with tests

```ts
import { assertThrows } from "jsr:@std/assert";
Deno.test({ permissions: { read: true } }, function execPathPerm() {
  console.log(Deno.permissions.querySync({ "name": "read" }));
  if (Deno.build.os === "linux") {
    assertThrows(
      () => {
        const a = Deno.readTextFileSync("/proc/net/dev");
        console.log(a);
      },
      Deno.errors.PermissionDenied,
      "Requires read access to <exec_path>, run again with the --allow-read flag",
    );
  }
});
```

if I run this on the main branch with deno test -A, it doesn't error as expected, seems somehow like even though I specified only read = true, the tests are passing allow-all permission in implicit way, so it doesn't throw

File: tests/unit/os_test.ts [hunk 1]
```diff
@@ -196,6 +197,37 @@ Deno.test({ permissions: { read: false } }, function execPathPerm() {
   );
 });
 
+Deno.test(async function execPathPerm() {
+  if (Deno.build.os !== "linux") return;
+  // This is hack to bypass a bug in deno test runner,
+  // Currently if you specify {read: true} permission, it will stil pass --allow-all (tests are run with deno test --allow-all) implicitly, so this test won't work
+  // The workaround is to spawn a deno executable with the needed permissions
+  // TODO(#25085): remove this hack when the bug is fixed
```
Comment (dsherret) [inline tests/unit/os_test.ts:205]: I'm not sure this is a bug. I would expect `--allow-all` + `--allow-read` to mean `--allow-all`.

File: tests/unit/os_test.ts [hunk 2]
```diff
@@ -196,6 +197,22 @@ Deno.test({ permissions: { read: false } }, function execPathPerm() {
...
+  function procRequiresAllowAll() {
+    assertThrows(
+      () => {
+        Deno.readTextFileSync("/proc/net/dev");
+      },
+      Deno.errors.PermissionDenied,
+      `PermissionDenied: Requires all access to "/proc/net/dev", run again with the --allow-all flag`,
+    );
+  },
+);
```
Comment (dsherret) [inline tests/unit/os_test.ts:214]: I tried it out and `cargo test js_unit_tests::os_test` runs fine with this, so I reverted to it.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

File: tests/unit/os_test.ts (hunk 2 above)
Comment (dsherret) [reply tests/unit/os_test.ts:213]: Hmmm... it was passing locally and now it's failing. I probably wasn't running it correctly. I'm looking into this.

Comment (dsherret) [reply tests/unit/os_test.ts:213]: I'm going to revert my changes and follow up in a separate pr

**Pull request merged**